### PR TITLE
Add #10020 [v100] move delete recent history groups

### PR DESCRIPTION
**Pull request by yoanarios**
Issue #10020 
Includes fix for #10469

- Add delete option to bottom toolbar for HistoryPanel main level
- Remove delete from tableview in HistoryPanel
- Quick search history fixes

### CODE REVIEW

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

File: Client/Application/ImageIdentifiers.swift [hunk 1]
```diff
@@ -32,6 +32,7 @@ public struct ImageIdentifiers {
     public static let key = "key"
     public static let libraryPanelHistory = "library-history"
     public static let libraryPanelSearch = "search"
+    public static let libraryPanelDelete = "action_delete"
```
Comment (nishant2718) [inline Client/Application/ImageIdentifiers.swift:35]: I believe we already have a `trashIcon`. Is this an image duplicate? 🤔 If it is, we should mark it in the TODO to address later and stick to one.

File: Storage/SQL/SQLiteHistory.swift [hunk 2]
```diff
@@ -621,6 +621,7 @@ extension SQLiteHistory: BrowserHistory {
                 historyFTS.rowid = hist.rowid
             WHERE historyFTS.title LIKE "%\(searchTerm)%" OR
                 historyFTS.url LIKE "%\(searchTerm)%"
+            ORDER BY local_modified DESC
```
Comment (nishant2718) [inline Storage/SQL/SQLiteHistory.swift:624]: I think I'll need to touch base with you about this one.

File: Client/Application/ImageIdentifiers.swift (hunk 1 above)
Comment (yoanarios) [reply Client/Application/ImageIdentifiers.swift:35]: I'm reusing one of the trashIcon we have in the app, I just added to the list of ImageIdentifiers

File: Storage/SQL/SQLiteHistory.swift (hunk 2 above)
Comment (yoanarios) [reply Storage/SQL/SQLiteHistory.swift:624]: Sure we sync on it, this was a request from Search design review they wanted to see the latest visits first

**Comment by nishant2718** [comment]
I'm seeing odd refresh behaviour after the search is performed. Will schedule some time to go over these.

**Comment by nishant2718** [comment]
Note: @yoanarios and I are aware of those strange behaviors and they're getting addressed in a different PR.

**Pull request merged**